hypershift/baremetal/agent.md

```diff
@@ -0,0 +1,502 @@
+image-registry                             4.9.21    True        False         False      61m     
+ingress                                    4.9.21    True        False         True       51m     The "default" ingress controller reports Degraded=True: DegradedConditions: One or more other status conditions indicate a degraded state: PodsScheduled=False (PodsNotScheduled: Some pods are not scheduled: Pod "router-default-7f9dc784cb-mlg9m" cannot be scheduled: 0/1 nodes are available: 1 node(s) didn't have free ports for the requested pod ports. Make sure you have sufficient worker nodes.), CanaryChecksSucceeding=False (CanaryChecksRepetitiveFailures: Canary route checks for the default ingress controller are failing)
+kube-apiserver                             4.9.21    True        False         False      4h27m   
+kube-controller-manager                    4.9.21    True        False         False      4h27m   
+kube-scheduler                             4.9.21    True        False         False      4h27m   
+kube-storage-version-migrator              4.9.21    True        False         False      61m     
+monitoring                                           False       True          True       45m     Rollout of the monitoring stack failed and is degraded. Please investigate the degraded status error.
+network                                    4.9.21    True        False         False      63m     
+node-tuning                                4.9.21    True        False         False      61m     
+openshift-apiserver                        4.9.21    True        False         False      4h27m   
+openshift-controller-manager               4.9.21    True        False         False      4h27m   
+openshift-samples                          4.9.21    True        False         False      60m     
+operator-lifecycle-manager                 4.9.21    True        False         False      4h26m   
+operator-lifecycle-manager-catalog         4.9.21    True        False         False      4h26m   
+operator-lifecycle-manager-packageserver   4.9.21    True        False         False      4h27m   
+service-ca                                 4.9.21    True        False         False      62m     
+storage                                    4.9.21    True        False         False      62m    
+
+oc get clusterversion --kubeconfig=${HOSTED}-kubeconfig
+NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
+version             False       True          4h28m   Unable to apply 4.9.21: some cluster operators have not yet rolled out
+~~~
+
+> **NOTE**: Some cluster operators are degraded because there is only a single worker and they require at least 2. However setting the `hostedcluster.spec.infrastructureAvailabilityPolicy: SingleReplica` configuration parameter disables the requirement and will make the clusters operator available with a single worker.
```

Comment:
How do you set it?

Reply:
According to the docs https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1alpha1.HostedCluster

```
apiVersion: hypershift.openshift.io/v1alpha1
kind: HostedCluster
metadata:
...
spec:
  infrastructureAvailabilityPolicy: SingleReplica
```